app/models/package_manager/go.rb

```diff
@@ -128,7 +130,18 @@ def self.project(name)
       # base package manager handles if the project is not present
       return nil if doc_html.text.blank?
 
-      { name: name, html: doc_html, overview_html: doc_html }
+      raw_project = { name: name, html: doc_html, overview_html: doc_html }
+
+      # pages on pkg.go.dev can be listed as 'package', 'module', or both. We only scrape Go Modules.
+      unless is_a_module?(raw_project: raw_project)
+        if !is_a_project?(raw_project: raw_project) && !is_a_command?(raw_project: raw_project) && !is_a_directory?(raw_project: raw_project) && defined?(Bugsnag)
+          # this is more for our record-keeping so we know what possible types there are.
+          Bugsnag.notify(UnknownGoType.new("Unknown Go project type (it is neither a package, a command or a module) for #{name}: #{page_type}"))
+        end
+        return nil
+      end
```

Comment:
@mikeyoung85 I basically moved the `is_a_module?` into a method that `module?` can use below, but I couldn't use `module?` itself here because there are a lot of actually legit Go Modules that are marked as "Not valid go.mod" on pkg.go.dev, so we should probably continue to ingest those